src/PublishPort.jsx

```diff
@@ -7,10 +8,47 @@ import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput"
 import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
 import { OutlinedQuestionCircleIcon, TrashIcon } from '@patternfly/react-icons';
 import cockpit from 'cockpit';
+import ipaddr from "ipaddr.js";
+
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import * as utils from './util.js';
 
 const _ = cockpit.gettext;
 
-export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
+export function validatePublishPort(value, key, validationFailed) {
+    const delta = { ...validationFailed };
+
+    switch (key) {
+    case "IP":
+        if (value && !ipaddr.isValid(value))
+            delta.IP = _("Must be a valid IP address");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231009-110933-d1e7ff0e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)